tests/datasets/conftest.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+from collections.abc import Iterator
+from typing import BinaryIO
+
+import pytest
+from pytest import MonkeyPatch
+
+
+class ContainerClient:
+    def __init__(self, account_url: str, container_name: str) -> None:
+        self.account_url = account_url
+        self.container_name = container_name
+
+    def list_blob_names(self, name_starts_with: str = "") -> Iterator[str]:
+        prefix = os.path.join(self.account_url, self.container_name)
+        for root, dirs, files in os.walk(prefix):
+            for file in files:
+                name = os.path.join(root, file).replace(prefix + os.sep, "", 1)
+                if name.startswith(name_starts_with):
+                    yield name
+
+    def download_blob(self, blob: str) -> BinaryIO:
+        path = os.path.join(self.account_url, self.container_name, blob)
+        # TODO: filehandle leak
+        f = open(path, "rb", buffering=0)
+        return f
```

Comment:
I don't really know what to do about this. On macOS, we're only allowed to open 256 files at a time. I don't _think_ any of our text data will exceed this, but I don't know how else to code this to mimic the actual behavior.